src/silo/prepare_dataset.cpp

```diff
@@ -227,104 +227,77 @@ void silo::partitionData(
    SPDLOG_INFO("Finished partitioning");
 }
 
-std::unordered_map<std::string, silo::common::Date> sortMetadataFile(
+std::unordered_map<std::string, size_t> sortMetadataFile(
    silo::preprocessing::MetadataReader& metadata_reader,
    silo::preprocessing::MetadataWriter& metadata_writer,
    const silo::preprocessing::PartitionChunk& chunk,
    const silo::SortChunkConfig& sort_chunk_config
 ) {
-   std::unordered_map<std::string, silo::common::Date> primary_key_to_date;
-
-   struct RowWithDate {
+   struct RowWithKeyDate {
       csv::CSVRow row;
+      std::string primary_key;
       silo::common::Date date;
+
+      RowWithKeyDate(csv::CSVRow& row, std::string&& primary_key, silo::common::Date date)
+          : row(row),
+            primary_key(primary_key),
+            date(date) {}
    };
-   std::vector<RowWithDate> rows;
+   std::vector<RowWithKeyDate> rows;
+   rows.reserve(chunk.size);
 
    for (auto& row : metadata_reader.reader) {
-      std::this_thread::sleep_for(std::chrono::nanoseconds(1));
```

Comment:
Ah, I noticed that it was not necessary when I ran it, as the iteration is slower because we read the entire row instead of just one column. But better add it back in I guess just to be safe